galore/plot.py

```diff
@@ -145,20 +151,22 @@ def plot_pdos(pdos_data, ax=None, total=True, show_orbitals=True,
         else:
             x_data = el_data['energy'] + offset
 
-        orbitals = list(el_data.keys())
-        orbitals.remove('energy')
-
-        for orbital in orbitals:
-            if total:
-                tdos += el_data[orbital]
-            else:
+        for orbital in orbitals: # orbitals defined above
+            if not total:
                 max_y = max(max_y, max(el_data[orbital]))
 
             if show_orbitals:
-                ax.plot(x_data, el_data[orbital],
-                        label="{0}: {1}".format(element, orbital),
+                color = next(ax._get_lines.prop_cycler)['color'] # get color to ensure
+                # matching fill
+                label = None if max(
+                    el_data[orbital]) < (legend_cutoff/100)*tmax else \
+                    "{0}: {1}".format(element, orbital)
+                ax.plot(x_data, el_data[orbital], color=color, label=label,
                         marker='', linestyle=next(linecycler))
 
+                if fill:
+                    ax.fill_between(x_data, el_data[orbital], color=color, alpha=alpha)
```

Comment:
It's probably fine, but can you verify that the fill_between will always put the shading _behind_ the lines? We might need to do some z-ordering to make it reliable.

Reply:
Have checked this is indeed the case, see https://stackoverflow.com/questions/28542933/fill-between-always-below-plot-lines and verified by looking at the output example plots in the PR:

<img width="1028" alt="image" src="https://user-images.githubusercontent.com/51478689/178999538-76603e0c-e1af-4c93-8c2b-c02db368ca2d.png">

Reply:
Looks nice indeed!